minesweeper/use-cases/markFlag/markFlag.ts

```diff
@@ -0,0 +1,47 @@
+import { BoardModel } from '@minesweeper/domain/Board.model';
+import { IBoardId, IPosition } from '@minesweeper/domain/commons.type';
+
+import { dataRepository } from '@minesweeper/infrastructure/data';
```

Comment:
The UseCases Layer can not depend of Infrastructure Layer, dataRepository should pass in props of markFlagUseCase function
```suggestion
import { IDataRepository } from '@minesweeper/domain/data.repository';
```